src/object/srv_obj.c

```diff
@@ -648,10 +776,51 @@ obj_local_rw(crt_rpc_t *rpc, struct ds_cont_hdl *cont_hdl,
 	if (opc_get(rpc->cr_opc) == DAOS_OBJ_RPC_UPDATE ||
 	    opc_get(rpc->cr_opc) == DAOS_OBJ_RPC_TGT_UPDATE) {
 		bulk_op = CRT_BULK_GET;
-		rc = vos_update_begin(cont->sc_hdl, orw->orw_oid,
-				      orw->orw_epoch, &orw->orw_dkey,
-				      orw->orw_nr, orw->orw_iods.ca_arrays,
-				      &ioh, dth);
+		if (oca->ca_resil == DAOS_RES_EC) {
+			unsigned int tgt_idx = orw->orw_oid.id_shard -
+						orw->orw_start_shard;
+
+			for (i = 0; i < orw->orw_nr; i++)
+				skip_list[i] = NULL;
+			if (tgt_idx >= oca->u.ec.e_p) {
+				rc = ec_data_target(tgt_idx - oca->u.ec.e_p,
+						    orw->orw_nr,
+						    orw->orw_iods.ca_arrays,
+						    oca, skip_list);
+			} else if (tgt_idx == 0) {
+				rc = ec_copy_iods(&iods,
+						  orw->orw_iods.ca_arrays,
+						  orw->orw_nr);
+				if (rc == 0) {
+					rc = ec_parity_target(tgt_idx,
+							      orw->orw_nr,
+							      iods, oca,
+							      skip_list);
+				}
+			} else {
+				rc = ec_parity_target(tgt_idx,
```

Comment:
similar to the comment below we could do something like this
else {
     tmp_iods = orw->orw_iods.ca_arrays;
     if (tdt_idx == 0) {
             ec_copy_iods(....);
             tmp_iods = iods;
     }

   rc = ec_parity_target(...tmp_iods);

That brings up a question though....is orw_nr the same for iods vs ca_arrays?

Reply:
The iods array is an exact copy of orw->orw_iods.ca_arrays.  Hence orw->nr is the number of entries for both.  I introduced the temporary variable, as suggested, and now make just one call to ec_parity_target, and one call to ec_obj_update. (I changed the copy's name from  iods to cpy_iods, and named the temporary tmp_iods.)